repositories/patches/rmw_implementation_library_path.patch

```diff
@@ -1,41 +1,58 @@
 diff --git a/rmw_implementation/src/functions.cpp b/rmw_implementation/src/functions.cpp
-index 41f164c..7407f77 100644
+index 892eb40..4316143 100644
 --- a/rmw_implementation/src/functions.cpp
 +++ b/rmw_implementation/src/functions.cpp
-@@ -46,35 +46,7 @@ static std::shared_ptr<rcpputils::SharedLibrary> g_rmw_lib = nullptr;
- std::shared_ptr<rcpputils::SharedLibrary>
- load_library()
- {
--  std::string env_var;
--  try {
--    env_var = rcpputils::get_env_var("RMW_IMPLEMENTATION");
--  } catch (const std::exception & e) {
--    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
--      "failed to fetch RMW_IMPLEMENTATION "
--      "from environment due to %s", e.what());
--    return nullptr;
--  }
--
--  if (env_var.empty()) {
--    env_var = STRINGIFY(DEFAULT_RMW_IMPLEMENTATION);
--  }
--
--  std::string library_path;
--  try {
--    library_path = rcpputils::find_library_path(env_var);
--  } catch (const std::exception & e) {
--    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
--      "failed to find shared library due to %s", e.what());
--    return nullptr;
--  }
--
--  if (library_path.empty()) {
--    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
--      "failed to find shared library '%s'",
--      env_var.c_str());
--    return nullptr;
+@@ -20,7 +20,6 @@
+ #include <stdexcept>
+ #include <string>
+ 
+-#include "ament_index_cpp/get_resources.hpp"
+ 
+ #include "rcutils/allocator.h"
+ #include "rcutils/format_string.h"
+@@ -82,6 +81,7 @@ load_library()
+   // 3. If that fails, try loading all other implementations available in turn
+   //    until one succeeds or we run out of options.
+ 
++#ifndef RMW_LIBRARY_PATH
+   std::string env_var;
+   try {
+     env_var = rcpputils::get_env_var("RMW_IMPLEMENTATION");
+@@ -105,25 +105,23 @@ load_library()
+     return ret;
+   }
+ 
+-  // OK, we failed to load the default RMW.  Fetch all of the ones we can
+-  // find and attempt to load them one-by-one.
+-  rmw_reset_error();
+-  const std::map<std::string, std::string> packages_with_prefixes = ament_index_cpp::get_resources(
+-    "rmw_typesupport");
+-  for (const auto & package_prefix_pair : packages_with_prefixes) {
+-    if (package_prefix_pair.first != "rmw_implementation") {
+-      ret = attempt_to_load_one_rmw(package_prefix_pair.first);
+-      if (ret != nullptr) {
+-        return ret;
+-      }
+-      rmw_reset_error();
+-    }
 -  }
+ 
+   // If we made it here, we couldn't find an rmw to load.
+ 
+   RMW_SET_ERROR_MSG("failed to load any RMW implementations");
+ 
++#else
 +  const std::string library_path = RMW_LIBRARY_PATH;
++  try {
++    return std::make_shared<rcpputils::SharedLibrary>(library_path.c_str());
++  } catch (const std::exception & e) {
++    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
++      "failed to load shared library '%s' due to %s",
++      library_path.c_str(), e.what());
++    return nullptr;
++  }
```

Comment:
In fact, we just need this part. I don't think we need `#ifndef RMW_LIBRARY_PATH` stuff.